go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -306,18 +312,37 @@ func (app *registryApplication) registerNode( // nolint: gocyclo
 		sc.Close()
 	}()
 
-	if isNewNode || isExpiredNode {
-		// Check that the entity has enough stake for this node registration.
-		if !params.DebugBypassStake {
-			if err = stakeCache.EnsureNodeRegistrationStake(newNode.EntityID, numEntityNodes+1); err != nil {
-				ctx.Logger().Error("RegisterNode: insufficient stake for new node",
-					"err", err,
-					"entity", newNode.EntityID,
-				)
-				return err
-			}
+	// Check that the entity has enough stake for this node registration.
+	var stakeAcc *stakingState.StakeAccumulatorCache
+	if !params.DebugBypassStake {
+		stakeAcc, err = stakingState.NewStakeAccumulatorCache(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to create stake accumulator cache: %w", err)
 		}
 
+		claim := stakeClaimForNode(newNode.ID)
+		var thresholds []staking.ThresholdKind
+		if newNode.HasRoles(node.RoleComputeWorker | node.RoleKeyManager) {
+			thresholds = append(thresholds, staking.KindCompute)
```

Comment:
The key manager doesn't get it's own threshold?  Shouldn't it?

Reply:
Yeah, good idea, I'll add it.

Reply:
Done.